src/main/resources/org/jenkinsci/plugins/scriptsecurity/scripts/ScriptApproval/script-approval.js

```diff
@@ -0,0 +1,236 @@
+function hideScript(hash) {
```

Comment:
Diff from original:

```javascript
@@ -1,4 +1,3 @@
-var mgr = <st:bind value="${it}"/>;
 function hideScript(hash) {
     document.getElementById('ps-' + hash).remove();
 }
@@ -56,14 +55,6 @@
     } else {
         document.getElementById('pendingClasspathEntries-none').style.display = 'none';
         Array.from(document.getElementById('pendingClasspathEntries').children).forEach(function(e){e.remove()});
-        /*
-           Create a list like:
-            <p id="pcp-${pcp.hash}">
-                <button class="approve" onclick="approveClasspathEntry('${pcp.hash}')">Approve</button> /
-                <button class="deny"    onclick="denyClasspathEntry('${pcp.hash}')">Deny</button> 
-                ${pcp.hash} (${pcp.path})
-            </p>
-         */
         pendingClasspathEntries.forEach(function(e) {
             var block = document.createElement('p');
             block.setAttribute('id', 'pcp-' + e.hash);
@@ -103,13 +94,6 @@
     } else {
         document.getElementById('approvedClasspathEntries-none').style.display = 'none';
         Array.from(document.getElementById('approvedClasspathEntries').children).forEach(function(e){e.remove()});
-        /*
-           Create a list like:
-            <p id="acp-${acp.hash}">
-                <button class="delete" onclick="denyApprovedClasspathEntry('${pcp.hash}')">Delete</button> 
-                ${acp.hash} (${acp.path})
-            </p>
-         */
         approvedClasspathEntries.forEach(function(e) {
             var block = document.createElement('p');
             block.setAttribute('id', 'acp-' + e.hash);
@@ -161,8 +145,92 @@
     });
 }
 
-window.addEventListener("load", function(){
+document.addEventListener('DOMContentLoaded', function() {
     mgr.getClasspathRenderInfo(function(r) {
         renderClasspaths(r);
     });
+
+    const approveScriptButtons = document.querySelectorAll('.ps-context button.approve');
+    approveScriptButtons.forEach(function (button) {
+        button.addEventListener('click', function () {
+            approveScript(button.dataset.hash);
+        });
+    });
+
+    const denyScriptButtons = document.querySelectorAll('.ps-context button.deny');
+    denyScriptButtons.forEach(function (button) {
+        button.addEventListener('click', function () {
+            denyScript(button.dataset.hash);
+        });
+    });
+
+    const deprecatedApprovedScriptsClearButton = document.querySelector('#deprecated-approvedScripts-clear button');
+    if (deprecatedApprovedScriptsClearButton) {
+        deprecatedApprovedScriptsClearButton.addEventListener('click', function () {
+            if (confirm('Really delete all deprecated approvals? Any existing scripts will need to be requeued and reapproved.')) {
+                mgr.clearDeprecatedApprovedScripts();
+                document.getElementById('deprecated-approvedScripts-clear').style.display = 'none';
+            }
+        });
+    }
+
+    const approvedScriptsClearButton = document.querySelector('#approvedScripts-clear button');
+    approvedScriptsClearButton.addEventListener('click', function () {
+        if (confirm('Really delete all approvals? Any existing scripts will need to be requeued and reapproved.')) {
+            mgr.clearApprovedScripts();
+        }
+    });
+
+    const approveSignatureButtons = document.querySelectorAll('.s-context button.approve');
+    approveSignatureButtons.forEach(function (button) {
+        button.addEventListener('click', function () {
+            approveSignature(button.dataset.signature, button.dataset.hash);
+        });
+    });
+
+    const aclApproveSignatureButtons = document.querySelectorAll('.s-context button.acl-approve');
+    aclApproveSignatureButtons.forEach(function (button) {
+        button.addEventListener('click', function () {
+            aclApproveSignature(button.dataset.signature, button.dataset.hash);
+        });
+    });
+
+    const denySignatureButtons = document.querySelectorAll('.s-context button.deny');
+    denySignatureButtons.forEach(function (button) {
+        button.addEventListener('click', function () {
+            denySignature(button.dataset.signature, button.dataset.hash);
+        });
+    });
+
+    const approvedSignaturesClearButton = document.querySelector('#approvedSignatures-clear button');
+    approvedSignaturesClearButton.addEventListener('click', function () {
+        if (confirm('Really delete all approvals? Any existing scripts will need to be rerun and signatures reapproved.')) {
+            clearApprovedSignatures();
+        }
+    });
+
+    const dangerousApprovedSignaturesClearButton = document.querySelector('#dangerousApprovedSignatures-clear button');
+    if (dangerousApprovedSignaturesClearButton) {
+        dangerousApprovedSignaturesClearButton.addEventListener('click', function () {
+            clearDangerousApprovedSignatures();
+        });
+    }
+
+    const deprecatedApprovedClasspathsClearButton = document.querySelector('#deprecated-approvedClasspaths-clear-btn button');
+    if (deprecatedApprovedClasspathsClearButton) {
+        deprecatedApprovedClasspathsClearButton.addEventListener('click', function () {
+            if (confirm('This will be scheduled on a background thread. You can follow the progress in the system log')) {
+                mgr.convertDeprecatedApprovedClasspathEntries();
+                document.getElementById('deprecated-approvedClasspaths-clear-btn').style.display = 'none';
+                document.getElementById('deprecated-approvedClasspaths-clear-spinner').style.display = '';
+            }
+        });
+    }
+
+    const approvedClasspathEntriesClearButton = document.querySelector('#approvedClasspathEntries-clear button');
+    approvedClasspathEntriesClearButton.addEventListener('click', function () {
+        if (confirm('Really delete all approvals? Any existing scripts using a classpath will need to be rerun and entries reapproved.')) {
+            clearApprovedClasspathEntries();
+        }
+    });
 });
```